packages/core/src/utils/set.ts

```diff
@@ -0,0 +1,66 @@
+export class CountableSet<K> {
```

Comment:
How about extending `Set`, override the `add` function to count, and introduce an additional function to get the counts?

Reply:
Do you mean something like this, right?
E.g:
```ts
export class CountableSet<K> extends Set {
  _map = new Map<K, number>()

  add(key: K) {
    super.add(key)
    this._map.set(key, (this._map.get(key) ?? 0) + 1)
    return this
  }

  getCount(key: K) {
    return this._map.get(key) ?? 0
  }
}
``` 
But in this case Set's values with be duplicated as Map's keys. Not sure if it is a problem 🤔

Reply:
and we also need `delete` method:
```ts
delete(key: K) {
  super.delete(key)
  return this._map.delete(key)
}
```

Reply:
Yes :)

Reply:
Can we use a simple getter instead of overriding Set methods ?

```ts
class MySet extends Set {
    get lastValue() {
        return [...this][this.size - 1]
    }
}
```